contracts/modules/SwapsImplSovrynSwapModule.sol

```diff
@@ -0,0 +1,109 @@
+pragma solidity 0.5.17;
+pragma experimental ABIEncoderV2;
+
+import "../swaps/connectors/SwapsImplSovrynSwapInternal.sol";
+import "../events/ModulesCommonEvents.sol";
+
+contract SwapsImplSovrynSwapModule is SwapsImplSovrynSwapInternal, ModulesCommonEvents {
+    /**
+     * @notice Empty public constructor.
+     * */
+    constructor() public {}
+
+    /**
+     * @notice Fallback function is to react to receiving value (rBTC).
+     * */
+    function() external {
+        revert("fallback not allowed");
+    }
+
+    /**
+     * @notice Set function selectors on target contract.
+     *
+     * @param target The address of the target contract.
+     * */
+    function initialize(address target) external onlyOwner {
+        address prevModuleContractAddress =
+            logicTargets[this.getSovrynSwapNetworkContract.selector];
+        _setTarget(this.getSovrynSwapNetworkContract.selector, target);
+        _setTarget(this.getContractHexName.selector, target);
+        _setTarget(this.swapsImplInternalExpectedRate.selector, target);
+        _setTarget(this.swapsImplInternalExpectedReturn.selector, target);
+        emit ProtocolModuleContractReplaced(
+            prevModuleContractAddress,
+            target,
+            "SwapsImplSovrynSwapModule"
+        );
+    }
+
+    /**
+     * Get the hex name of a contract.
+     * @param source The name of the contract.
+     * */
+    function getContractHexName(string memory source) public pure returns (bytes32 result) {
+        return _getContractHexName(source);
+    }
+
+    /**
+     * Look up the Sovryn swap network contract registered at the given address.
+     * @param sovrynSwapRegistryAddress The address of the registry.
+     * */
+    function getSovrynSwapNetworkContract(address sovrynSwapRegistryAddress)
+        public
+        view
+        returns (ISovrynSwapNetwork)
+    {
+        return _getSovrynSwapNetworkContract(sovrynSwapRegistryAddress);
+    }
+
+    /**
+     * @notice Get the expected rate for 1 source token when exchanging the
+     *   given amount of source tokens.
+     *
+     * @param sourceTokenAddress The address of the source token contract.
+     * @param destTokenAddress The address of the destination token contract.
+     * @param sourceTokenAmount The amount of source tokens to get the rate for.
+     * */
+    function swapsImplInternalExpectedRate(
```

Comment:
```suggestion
    function swapsImplExpectedRate(
```

Reply:
updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/530/commits/0334bfa7656f28d673a27380fd05475bcc0b6524)